packages/shared/src/api/postsApi.ts

```diff
@@ -102,23 +104,33 @@ export async function getPostReference({
     replyId ? '/' + replyId : ''
   }`;
   const data = await subscribeOnce<ub.Said>({ app: 'channels', path }, 3000);
-  const post = toPostReference(data);
+  const post = await toPostReference(data);
   // The returned post id can be different than the postId we requested?? But the
   // post is going to be requested by the original id, so set manually :/
   post.id = postId;
   return post;
 }
 
-function toPostReference(said: ub.Said) {
+async function toPostReference(said: ub.Said) {
   const channelId = said.nest;
+  const hiddenPosts = await db.getHiddenPosts();
   if ('reply' in said.reference) {
     return toPostReplyData(
       channelId,
       said.reference.reply['id-post'],
-      said.reference.reply.reply
+      said.reference.reply.reply,
+      hiddenPosts.some(
```

Comment:
Is this stuff in here to prevent overwriting the `hidden` field on posts? I'm wondering if we could skip doing it here, and just prevent `hidden` from being overwritten on insert (via `onConflictDoUpdate`). If that doesn't work, can we move the db wrangling either into sync or into the query?